### PR TITLE
Feat/nurse management

### PR DESCRIPTION
**Pull request by enjaytarigan**
Done:

-  Added create nurse user api

Changed:

- Change ValidateNIP method to entity's public function
- Added validation for user's identity image url
- Added auth middleware

Todo:
- Add several errors handling

### CODE REVIEW

File: w3/domain/entity/user_entity.go [hunk 1]
```diff
@@ -16,18 +20,19 @@ const (
...
-	Password            string       `json:"password"`
-	Role                string       `json:"role"`
-	IdentityCardScanImg string       `json:"identity_card_scan_img"`
-	CreatedAt           time.Time    `json:"created_at"`
-	UpdatedAt           time.Time    `json:"updated_at"`
-	DeletedAt           sql.NullTime `json:"deleted_at"`
+	ID                  uuid.UUID      `json:"id"`
+	NIP                 string         `json:"nip"`
+	Name                string         `json:"name"`
+	Password            sql.NullString `json:"password"`
```
Comment (shamir92) [inline w3/domain/entity/user_entity.go:26]: prefer string aja om. 
@enjaytarigan

File: w3/domain/entity/user_entity.go [hunk 2]
```diff
@@ -73,3 +78,31 @@ func (e *User) ValidateNIP(nip string, role UserRole) bool {
 
 	return true
 }
+
+func ValidateIdentityCardScanImageURL(rawUrl string) bool {
```
Comment (shamir92) [inline w3/domain/entity/user_entity.go:82]: @enjaytarigan  
bang, apakah lebih baik pakai playground validator ?

File: w3/domain/usecase/user_it_usecase.go [hunk 3]
```diff
@@ -58,7 +59,10 @@ func (u *userITUsecase) RegisterUserIT(userITRequest UserITRegisterRequest) (Use
 	}
 	user.Name = userITRequest.Name
 	user.NIP = userNip
-	user.Password = hashedPassword
+	user.Password = sql.NullString{
```
Comment (shamir92) [inline w3/domain/usecase/user_it_usecase.go:62]: ini juga

File: w3/domain/usecase/user_it_usecase.go [hunk 4]
```diff
@@ -105,7 +109,7 @@ func (u *userITUsecase) LoginUserIT(request UserITLoginRequest) (UserITLoginResp
 	if err != nil {
 		return UserITLoginResponse{}, err
 	}
-	if !u.bcryptHelper.Compare(user.Password, request.Password) {
+	if !u.bcryptHelper.Compare(user.Password.String, request.Password) {
```
Comment (shamir92) [inline w3/domain/usecase/user_it_usecase.go:112]: ini berubah karena sql.nullstring

**Review by shamir92 — CHANGES_REQUESTED** [review]
@enjaytarigan  
ada comment ya bang.

File: w3/domain/usecase/user_it_usecase.go (hunk 3 above)
Comment (enjaytarigan) [reply w3/domain/usecase/user_it_usecase.go:62]: It's not about go-validator, but as we know that password is nullable (ERD)

File: w3/domain/usecase/user_it_usecase.go (hunk 4 above)
Comment (enjaytarigan) [reply w3/domain/usecase/user_it_usecase.go:112]: That's the way to access the real value in sql.NullString type.

**Review by shamir92 — APPROVED** [review]
lgtm

**Pull request merged**